### PR TITLE
fix #26890 avoid duplicate overlay drivers in info

### LINKED ISSUE #26890: docker/master shows two overlay network plugins in `docker info`
**Description**

**Steps to reproduce the issue:**
1. build the docker/master code 
2. use the newly built dockerd to run docker daemon
3. execute `docker info`

**Describe the results you received:**
```
Plugins:
 Volume: local
 Network: bridge host null overlay overlay
```
Got two overlays.

**Describe the results you expected:**
```
Plugins:
 Volume: local
 Network: bridge host null overlay 
```
Only one overlay.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
root@10-11-26-89:~/gocode/src/github.com/docker/docker# docker version
Client:
 Version:      1.13.0-dev
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   136ce0f
 Built:        Sun Sep 25 11:16:12 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.0-dev
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   98764fe
 Built:        Sun Sep 25 10:08:30 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
root@10-11-26-89:~/gocode/src/github.com/docker/docker# docker info
Containers: 10
 Running: 0
 Paused: 0
 Stopped: 10
Images: 72
Server Version: 1.13.0-dev
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 72
 Dirperm1 Supported: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host null overlay overlay
Swarm: active
 NodeID: c8whwhavxd87tqcwgezydf1ve
 Is Manager: true
 ClusterID: 1cxjyiz0stofeu7ia1t9qfb3q
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 10.11.26.89
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 3.13.0-63-generic
Operating System: Ubuntu 14.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.955 GiB
Name: 10-11-26-89
ID: 62DZ:LXFO:EOE7:GVJD:32AV:UXF4:FH4B:M4UM:FTJO:7XC7:2QOZ:KB7B
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 1383
 Goroutines: 1476
 System Time: 2016-09-25T04:19:18.18305008-07:00
 EventsListeners: 0
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

### PR DESCRIPTION
**Pull request by allencloud**
fixes #26890 

**\- What I did**
Check overlay whether it is in pluginMap, if not add it, otherwise not.

**\- How I did it**

**\- How to verify it**

**\- Description for the changelog**

**\- A picture of a cute animal (not mandatory but encouraged)**

Signed-off-by: allencloud allen.sun@daocloud.io

### CODE REVIEW

File: daemon/network.go [hunk 1]
```diff
@@ -372,7 +372,10 @@ func (daemon *Daemon) GetNetworkDriverList() []string {
 		}
 	}
 	// TODO : Replace this with proper libnetwork API
-	pluginList = append(pluginList, "overlay")
+	if !pluginMap["overlay"] {
+		pluginList = append(pluginList, "overlay")
+		pluginMap["overlay"] = true
```
Comment (thaJeztah) [inline daemon/network.go]: Perhaps move these two lines above the for-loop above, and just initialize pluginList, and pluginMap there;

```go
pluginList := []string{"overlay"}
pluginMap := map[string]bool{"overlay":true}
```

Comment (thaJeztah) [inline daemon/network.go]: (and keep the "todo")

**Review by thaJeztah — CHANGES_REQUESTED** [review]

File: daemon/network.go (hunk 1 above)
Comment (allencloud) [inline daemon/network.go]: Updated, PTAL @thaJeztah

File: daemon/network.go [hunk 2]
```diff
@@ -357,8 +357,8 @@ func (daemon *Daemon) DisconnectContainerFromNetwork(containerName string, netwo
 // GetNetworkDriverList returns the list of plugins drivers
 // registered for network.
 func (daemon *Daemon) GetNetworkDriverList() []string {
-	pluginList := []string{}
-	pluginMap := make(map[string]bool)
+	pluginList := []string{"overlay"}
+	pluginMap := map[string]bool{"overlay": true}
```
Comment (thaJeztah) [inline daemon/network.go]: perhaps, while we're at it, we could move this below the `if !daemon.NetworkControllerEnabled()` below, but let me know what you think. Also may want to wait until other people reviewed this

Comment (thaJeztah) [inline daemon/network.go]: Oh, and move the "todo" here (I see my comment may have been confusing :blush:

Comment (allencloud) [inline daemon/network.go]: Updated again. I think move it below is better, and also moved TODO here 😄

File: daemon/network.go [hunk 3]
```diff
@@ -357,12 +357,15 @@ func (daemon *Daemon) DisconnectContainerFromNetwork(containerName string, netwo
...
-	pluginMap := make(map[string]bool)
-
 	if !daemon.NetworkControllerEnabled() {
 		return nil
 	}
+
+	pluginList := []string{"overlay"}
+	pluginMap := map[string]bool{"overlay": true}
+
+	// TODO: Replace this with proper libnetwork API
```
Comment (anusha-ragunathan) [inline daemon/network.go]: TODO should be moved to L363

Comment (anusha-ragunathan) [inline daemon/network.go]: Rest of the change LGTM.

Comment (allencloud) [inline daemon/network.go]: Updated. Thanks for your review 😄

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**